bin/grading_done.py

```diff
@@ -57,43 +53,79 @@ def main():
                 proc = psutil.Process(pid)
                 if 'hwcron' == proc.username():
                     if (len(proc.cmdline()) >= 2 and
-                        proc.cmdline()[1] == os.path.join(SUBMITTY_INSTALL_DIR,"bin","submitty_grading_scheduler.py")):
+                        proc.cmdline()[1] == os.path.join(SUBMITTY_INSTALL_DIR,"bin","submitty_autograding_shipper.py")):
                         num_procs+=1
             except psutil.NoSuchProcess:
                 pass
 
-        # remove 2 from the count...  each worker is forked from the
-        # initial process, and there's a helper process for the
-        # observer too
-        num_procs-=2
+        # remove 1 from the count...  each worker is forked from the
+        # initial process
+        num_procs-=1
 
         if num_procs <= 0:
-            print ("WARNING: No matching submitty_grading_scheduler.py processes!")
+            print ("WARNING: No matching submitty_autograding_shipper.py processes!")
             num_procs = 0
 
         done = True
-        batch_queue = os.listdir(BATCH_QUEUE)
-        grading_batch_queue = list(filter(lambda x: x.startswith("GRADING"), batch_queue))
-        if len(batch_queue) != 0:
-            done = False
 
         print("GRADING PROCESSES:{:3d}       ".format(num_procs), end="")
 
-        if os.access(INTERACTIVE_QUEUE, os.R_OK):
+        if os.access(GRADING_QUEUE, os.R_OK):
             # most instructors do not have read access to the interactive queue
-            interactive_queue = os.listdir(INTERACTIVE_QUEUE)
-            grading_interactive_queue = list(filter(lambda x: x.startswith("GRADING"), interactive_queue))
-            print("INTERACTIVE todo:{:3d} ".format(len(interactive_queue)-len(grading_interactive_queue)), end="")
-            if len(grading_interactive_queue) == 0:
+
+            files = glob.glob(os.path.join(GRADING_QUEUE, "*"))
+            interactive_count = 0
+            interactive_grading_count = 0
+            regrade_count = 0
+            regrade_grading_count = 0
+
+            for full_path_file in files:
```

Comment:
Use `for full_path_file in glob.iglob(os.path.join(GRADING_QUEUE, '*'))` here as we don't need the full list, just an iterator through it.